src/shared/sleep-config.c

```diff
@@ -108,77 +111,132 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
         return 0;
 }
 
+/* Get the list of battery */
+static int get_sd_device_battery(sd_device_enumerator *e) {
+        int r;
+
+        r = sd_device_enumerator_new(&e);
+        if (r < 0)
+                return log_oom();
+
+        r = sd_device_enumerator_add_match_property(e, "POWER_SUPPLY_TYPE", "Battery");
```

Comment:
Not only the property, please add subsystem match for `"power_supply"` to improve performance.

Reply:
yes agreed. Fixed.